File: pynet/examples/run_client.py
#!/usr/bin/python

from pynet import connection

conn = connection.Connection()
assert conn.connect('localhost', 1337)
s = conn.send_and_receive('what the fuck')
print 'got back: [%s]' % s
assert s == 'WHAT THE FUCK'
s = conn.send_and_receive('righto')
print 'got back: [%s]' % s
assert s == 'RIGHTO'

File: net/examples/run_client.cc
#include <cassert>
#include <string>
#include "../connection.h"

using std::string;

int main() {
  Connection conn;
  assert(conn.Connect("localhost", 1337, 1024));

  string in = "what the fuck are you doing?";
  string out;
  assert(conn.SendAndReceive(in, &out));
  printf("got = [%s]\n", out.c_str());

  in = out;
  assert(conn.SendAndReceive(in, &out));
  printf("got = [%s]\n", out.c_str());
}
